.github/workflows/tests.yaml

```diff
@@ -19,4 +19,8 @@ jobs:
     - name: Get dependencies
       run: go get -v -t -d ./...
     - name: Run tests
-      run: go test main_test.go main.go
+      run: go test -coverprofile=cover.out main_test.go main.go
+    - name: Install Vim
+      run: apt-get update; apt-get install -y vim;
+    - name: Go Beautiful HTML Coverage
+      uses: gha-common/go-beautiful-html-coverage@v1
```

Comment:
When trying to push to the cover branch github actions needs the write permission which only @avelino has access to give.

Reply:
by default the branch is `cover`, I believe we need to bring the current branch for it to work - we don't have the `cover` branch for this flow.

```
    # The branch to checkout or create and push coverage to.
    # Default: 'cover'
    branch: ''
```

Reply:
I tested on my fork https://github.com/phanirithvij/awesome-go/pull/1
I had to enable github pages.

Now thinking about it, we already have github pages enabled for this repo, but can we enable one more github pages for the `cover` branch?

Reply:
> I believe we need to bring the current branch for it to work - we don't have the cover branch for this flow.

The action creates the branch for you. Letting the action create the branch and do a first time setup is the recommended way to do it. This is my suggestion:

- Let the action run in this PR, it will run all the setup code and create the branch.
- Turn on github pages and target `cover`
- Test that the links work and everything is kosher
- Merge!

Reply:
>Now thinking about it, we already have github pages enabled for this repo, but can we enable one more github pages for the cover branch?

if you already have github pages enabled for this repo, it complicates things. Let me think about it.

Reply:
GitHub Pages is active, but we don't use it to host the site - it's active because it was used in the past, today we don't use it

I just changed the default branch of GitHub Pages to the `cover` branch and root folder
<img width="756" alt="Screenshot 2024-05-24 at 15 04 19" src="https://github.com/avelino/awesome-go/assets/31996/b662d94d-fa05-4f16-964a-aef1ad04a932">

Reply:
> But our repo's contributions are rarely changing go code. @kilianc is it possible to detect only if go code changes and run this step?

Changes to the go files are not the only thing that affects the coverage, but maybe it's acceptable here. I configured [verify-changed-files](https://github.com/marketplace/actions/verify-changed-files#using-the-contains-function) to ignore all `.md` file changes because that's what most of the PRs here are on.

I kinda liked having it on because I took as an opportunity for folks to educate themselves about code coverage and copy this process in their own project, but I totally understand that there are already multiple automated comments on each PR.